Add source freshness examples dbt_project.yml file + update Combining configs (#7439)
fishtown-analytics/docs.getdbt.com: website/docs/reference/define-configs.md

## Combining configs

> Most configurations are "clobbered" or overwritten when applied hierarchically. Whenever a more specific value is available, it will completely replace the less specific value. Note that a few configs have different merge behavior:

Reviewer: I think it may be beneficial to add some specificity as to why the language "clobbered" differs from us saying just "overwritten"
```suggestion
Most configurations are "clobbered," or overwritten in unexpected ways, when applied hierarchically. Whenever a more specific value is available, it will completely replace the less specific value. Note that a few configs have different merge behavior:
```

> Most configurations are "clobbered," or overwritten in unexpected ways, when applied hierarchically. Whenever a more specific value is available, it will completely replace the less specific value. Note that a few configs have different merge behavior:

Reviewer 2: > or overwritten in unexpected ways

This is odd phrasing. It is expected & desirable behavior that less-specific configs are clobbered by more-specific configs. Then, the exceptions to this behavior are listed below.

> - [`tags`](/reference/resource-configs/tags) are additive. If a model has some tags configured in `dbt_project.yml`, and more tags applied in its `.sql` file, the final set of tags will include all of them.

Reviewer 3: ```suggestion
- [`tags`](/reference/resource-configs/tags) are additive. If a model has some tags configured in `dbt_project.yml`, and more tags are applied in its `.sql` file, the final set of tags will include all of them.
```

- [`meta`](/reference/resource-configs/meta) dictionaries are merged (a more specific key-value pair replaces a less specific value with the same key).
- [`pre-hook` and `post-hook`](/reference/resource-configs/pre-hook-post-hook) are also additive.

> - When using the [`freshness`](/reference/resource-configs/freshness) config for sources, note the following:

Reviewer 2: This list is about rules for clobbering / merging configurations that are inherited from multiple levels.

It is appropriate to have a bullet in this list about `freshness`, in order to say that it reflects the same behavior as `meta` - whereby dictionaries are merged (key-value pairs)

Otherwise, the `freshness` config observes / reflects these general patterns:
1. Node-level configs (more specific) clobber project-level configs (less specific)
2. For sources, table-level configs (more specific) clobber source-level configs (less specific)
3. The root project's configuration (in `dbt_project.yml`) clobbers configuration within package files, so that users ca control the behavior of packages they are installing via `dbt deps` without needing to edit the code in those package files directly (this would be an anti-pattern)

I think it would be more helpful to document ^those general patterns here, rather than talking about `freshness` configs specifically, because that makes it seem like `freshness` is an exception and that this behavior only to `freshness`

    - The `freshness` config in the `schema.yml` file overrides the `freshness` config in the `dbt_project.yml` file. 
    - A more specific `dbt_project.yml` file overrides a `dbt_project.yml` file with a less specific configuration. 

>     - In a `schema.yml` file, more specific `freshness` config merges with less specific `freshness` config.

Reviewer 3: ```suggestion
    - In a `schema.yml` file, more specific `freshness` configs are merged with less specific `freshness` configs.
```
